AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -744,6 +744,26 @@ - (void)__setNeedsLayout
   }
 }
 
+- (void)__setSafeFrame:(CGRect)rect
+{
+  BOOL useLayer = (_layer && ASDisplayNodeThreadIsMain());
```

Comment:
Should this lock the property lock? Otherwise should it be documented that it must always be called within the lock?